experimental/InjectiveResolutions/src/LocalCohomology.jl

```diff
@@ -0,0 +1,642 @@
+# data structures
+struct SectorLC
+  A::Vector{Vector{Int}}
+  sector::Polyhedron
+  index_vector::Vector{Int}
+  H::Generic.QuotientModule
+end
+mutable struct SectorPartitionLC
+  M::SubquoModule{<:MonoidAlgebraElem}
+  i::Int
+  I::MonoidAlgebraIdeal
+  sectors::Vector{SectorLC}
+  maps::Vector{Tuple{SectorLC,SectorLC,Generic.ModuleHomomorphism}}
+
+  function SectorPartitionLC(M::SubquoModule{<:MonoidAlgebraElem}, i::Int, I::MonoidAlgebraIdeal)
+    return new(
+      M,
+      i,
+      I,
+      Vector{SectorLC}(),
+      Vector{Tuple{SectorLC,SectorLC,Generic.ModuleHomomorphism}}(),
+    )
+  end
+end
+
+@doc raw"""
+    is_zero(s::SectorLC)
+
+Test if local cohomology is zero on a sector.
+"""
+function is_zero(s::SectorLC)
+  return dim(s.H) == 0
+end
+
+@doc raw"""
+    is_zero(S::SectorPartitionLC)
+
+Given a sector partition of a local cohomology module, check if the local cohomology module is zero.
+"""
+function is_zero(S::SectorPartitionLC)
+  return all([dim(s.H) == 0 for s in S.sectors])
```

Comment:
```suggestion
  return all(dim(s.H) == 0 for s in S.sectors)
```
... avoids an allocation of a vector for storing the intermediate result, can perform short circuiting (if the check fails for one `s` it can stop immediately)